course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -74,61 +56,632 @@ describe('Unit testing PeopleController', function() {
+            var memberDeferred = $q.defer();
+            // promises returned from lookupPeople, already resolved
+            var peoplePromises = [$q.resolve([]), $q.resolve([])];
+
+            // set up spies on methods addPeopleToCourse calls out to
+            spyOn(scope, 'lookupCourseMembers')
+                .and.returnValue(memberDeferred.promise);
+            spyOn(scope, 'lookupPeople').and.returnValue(peoplePromises);
+
+            // this is the one we're making sure doesn't get called
+            spyOn(scope, 'addNewMember');
+
+            // run it
+            scope.addPeopleToCourse(searchTermList);
+            scope.$digest();
+
+            // make sure it wasn't called
+            expect(scope.addNewMember).not.toHaveBeenCalled();
+
+            // now resolve the memberDeferred
+            memberDeferred.resolve([]);
+            scope.$digest();
+
+            // and affirm it was called, so we know it's not just a broken test
+            expect(scope.addNewMember).toHaveBeenCalledTimes(2);
+        });
+
         it('won\'t bother waiting for profile lookups or POSTing new members ' +
-            'if course membership lookup fails');
+                'if course membership lookup fails', function() {
+            // promise returned from lookupCourseMembers
+            var memberPromise = $q.reject(new Error());
+            // promises returned from lookupPeople
+            var peoplePromises = [$q.defer().promise, $q.defer().promise];
+
+            // set up spies on methods addPeopleToCourse calls out to
+            spyOn(scope, 'lookupCourseMembers').and.returnValue(memberPromise);
+            spyOn(scope, 'lookupPeople').and.returnValue(peoplePromises);
+
+            // this is the one we're making sure doesn't get called
+            spyOn(scope, 'addNewMember');
+
+            // and this is the one that should
+            spyOn(scope, 'showAddNewMemberResults');
+
+            // run it
+            scope.addPeopleToCourse(searchTermList);
+            scope.$digest();
+
+            // make sure the POST wasn't done
+            expect(scope.addNewMember).not.toHaveBeenCalled();
+
+            // and that the results were updated, despite the profile lookups
+            // being unresolved.
+            expect(scope.showAddNewMemberResults).toHaveBeenCalled();
+        });
     });
+
     describe('addNewMember', function() {
-        it('won\'t add person if they already have a course enrollment');
-        it('won\'t add if they could not be found via /people lookup');
-        it('won\'t add if multiple profiles returned by /people lookup');
-        it('adds person if not in course and single profile available');
+
+        var personResult = [];
+        var members = {};
+
+        beforeEach(function() {
+            controller = $controller('PeopleController', {$scope: scope });
+            //var members = {};
+            members["12345678"] = [{
```

Comment:
Hmm, does this mean we're resetting these each time? If so, we could collapse this into a single definition, e.g.
```
members = {
  12345678: [{ ... }],
  43215678: [{ ... }]
}
```
If we don't edit any of the values in the tests, we could move it out of the `beforeEach()`.